File: server/aoNewsletter2/Models/View/NewsletterViewModel.vb

Imports Contensive.Addons.Newsletter.Controllers
Imports Contensive.BaseClasses


Namespace Models.View
    Public Class NewsletterViewModel
        Inherits DesignBlockViewBaseModel
        '
        Public Property legacyNewsletter As String
        '
        '====================================================================================================
        ''' <summary>
        ''' Populate the view model from the entity model
        ''' </summary>
        ''' <param name="cp"></param>
        ''' <param name="settings"></param>
        ''' <returns></returns>
        Public Overloads Shared Function create(cp As CPBaseClass, settings As Models.Db.NewsletterModel, legacyNewsletter As String) As NewsletterViewModel
            Try
                '
                ' -- base fields
                Dim result = DesignBlockViewBaseModel.create(Of NewsletterViewModel)(cp, settings)
                '
                ' -- custom
                result.legacyNewsletter = legacyNewsletter
                '
                Return result
            Catch ex As Exception
                cp.Site.ErrorReport(ex)
                Return Nothing
            End Try
        End Function
    End Class

End Namespace

File: server/aoNewsletter2/Models/View/NewsletterViewModel.cs
using System;
using Contensive.BaseClasses;

namespace Contensive.Addons.Newsletter.Models.View {
    public class NewsletterViewModel : DesignBlockViewBaseModel {
        // 
        public string legacyNewsletter { get; set; }
        // 
        // ====================================================================================================
        /// <summary>
        /// Populate the view model from the entity model
        /// </summary>
        /// <param name="cp"></param>
        /// <param name="settings"></param>
        /// <returns></returns>
        public static NewsletterViewModel create(CPBaseClass cp, Db.NewsletterModel settings, string legacyNewsletter) {
            try {
                // 
                // -- base fields
                var result = create<NewsletterViewModel>(cp, settings);
                // 
                // -- custom
                result.legacyNewsletter = legacyNewsletter;
                // 
                return result;
            } catch (Exception ex) {
                cp.Site.ErrorReport(ex);
                return null;
            }
        }
    }

}